redash/tasks/queries.py

```diff
@@ -245,14 +246,33 @@ def truncate_long_string(original_str, max_length):
 
 
 @celery.task(name="redash.tasks.update_sample")
-def update_sample(data_source_id, table_name, table_id):
+def update_sample(data_source_id, table_name, table_id, sample_updated_at):
     """
     For a given table, find look up a sample row for it and update
     the "example" fields for it in the column_metadata table.
     """
     logger.info(u"task=update_sample state=start table_name=%s", table_name)
     start_time = time.time()
     ds = models.DataSource.get_by_id(data_source_id)
+
+    # Look at the first example in the persisted columns.
+    # If this is *not* empty AND sample_updated_at is recent, don't update sample
+    persisted_columns = ColumnMetadata.query.filter(
+        ColumnMetadata.exists.is_(True),
+        ColumnMetadata.table_id == table_id,
+    ).options(load_only('id', 'name', 'example'))
+
+    DAYS_AGO = (
+        utils.utcnow() - datetime.timedelta(days=settings.SCHEMA_SAMPLE_REFRESH_FREQUENCY_DAYS))
+
+    first_column = persisted_columns.first()
+
+    if (first_column and
+        sample_updated_at and
+        first_column.example and
+        parser.parse(sample_updated_at) > DAYS_AGO):
+        return
```

Comment:
Let's leave a code comment what this does so it's easier to decipher later, please. I also wonder if we should log this so we know what's going on.